File: examples/polyglot/hello.c
#include <stdio.h>

struct Greeter {
    const char* message;
};

void say_hello(struct Greeter* g) {
    printf("%s\n", g->message);
}

int main() {
    struct Greeter g = { "Hello C" };
    say_hello(&g);
    return 0;
}

File: examples/polyglot/hello.rs
struct Greeter {
    message: String,
}

impl Greeter {
    fn new(msg: &str) -> Greeter {
        Greeter { message: msg.to_string() }
    }

    fn say_hello(&self) {
        println!("{}", self.message);
    }
}

fn main() {
    let g = Greeter::new("Hello Rust");
    g.say_hello();
}
